tests/lib/rules/no-arrow-functions-in-watch.js

```diff
@@ -0,0 +1,196 @@
+      code: `
+        export default {
+          watch: {
+            foo() {},
+            bar() {}
+          },
+        }
+      `
+    },
+    {
+      filename: 'test.vue',
+      code: `
+        export default {
+          watch: {
+            foo: function() {},
+            bar: function() {}
+          },
+        }
+      `
+    },
+    {
+      filename: 'test.vue',
+      code: `
+      export default {
+        data: {
+          a: 1,
+          b: 2,
+          c: 3,
+          d: 4,
+          e: {
+            f: {
+              g: 5
+            }
+          }
+        },
+        watch: {
+          a: function (val, oldVal) {
+            console.log('new: %s, old: %s', val, oldVal)
+          },
+          b: 'someMethod',
+          c: {
+            handler: function (val, oldVal) {},
+            deep: true
+          },
+          d: {
+            handler: 'someMethod',
+            immediate: true
+          },
+          e: [
+            'handle1',
+            function handle2 (val, oldVal) {},
+            {
+              handler: function handle3 (val, oldVal) {},
+              /* ... */
+            }
+          ],
+          'e.f': function (val, oldVal) { /* ... */ }
+        }
+      }`
+    }
+  ],
+  invalid: [
+    {
+      filename: 'test.vue',
+      code: `
+      export default {
+        watch: {
+          foo: () => {}
+        },
+      }`,
+      errors: ['You should not use an arrow function to define a watcher.']
+    },
+    {
+      filename: 'test.vue',
+      code: `
+      export default {
+        watch: {
+          foo() {},
+          bar: () => {}
+        }
+      }`,
+      errors: ['You should not use an arrow function to define a watcher.']
+    },
+    {
+      filename: 'test.vue',
+      code: `
+      export default {
+        watch: {
+          foo: function() {},
+          bar: () => {}
+        }
+      }`,
+      errors: ['You should not use an arrow function to define a watcher.']
+    },
+    {
+      filename: 'test.vue',
+      code: `
+      export default {
+        data: {
+          a: 1,
+          b: 2,
+          c: 3,
+          d: 4,
+          e: {
+            f: {
+              g: 5
+            }
+          }
+        },
+        watch: {
+          a: (val, oldVal) => {
+            console.log('new: %s, old: %s', val, oldVal)
+          },
+          b: 'someMethod',
+          c: {
+            handler: function (val, oldVal) {},
+            deep: true
+          },
+          d: {
+            handler: 'someMethod',
+            immediate: true
+          },
+          e: [
+            'handle1',
+            function handle2 (val, oldVal) {},
+            {
+              handler: function handle3 (val, oldVal) {},
+              /* ... */
+            }
+          ],
+          'e.f': function (val, oldVal) { /* ... */ }
+        }
+      }`,
+      errors: ['You should not use an arrow function to define a watcher.']
```

Comment:
Could you check the location with some test cases?

```suggestion
      errors: [
        {
          message: 'You should not use an arrow function to define a watcher.',
          line: ?
        }
      ]
```